### PR TITLE
Improve window documentation

### PR DESCRIPTION
**Pull request by liyuqian**
To reflect Matt Carroll's recent (02/06/2019) email about promoting
`WidgetsBinding.instance.window` and demoting `ui.window`.

I have to manually search for Matt's email several times when I need
to access the Window object. Having it in the documentation would
help me do that much faster.

### CODE REVIEW

File: lib/ui/window.dart [hunk 1]
```diff
@@ -515,8 +515,12 @@ class Locale {
 
 /// The most basic interface to the host operating system's user interface.
 ///
+/// It exposes the size of the display, the core scheduler API, the input event
+/// callback, the graphics drawing API, and other such core services.
+///
 /// There is a single Window instance in the system, which you can
-/// obtain from the [window] property.
+/// obtain from `WidgetsBinding.instance.window`, or `window` in `dart:ui` if
+/// `WidgetsBinding` is unavailable.
```
Comment (matthew-carroll) [inline lib/ui/window.dart]: I think we really want to incentivize the use of the singleton though, right? If `window` is imported directly from `dart:ui` then that usage cannot be dependency injected, can it? If I'm remembering that right then we should probably use some very strong language to push people towards the binding, if at all possible. What do you think?

Comment (matthew-carroll) [reply lib/ui/window.dart]: I see below that you elaborated on this. I would suggest including a little bit of the seriousness of the issue in this reference, too.

Comment (liyuqian) [reply lib/ui/window.dart]: I've tried to make it more serious and strong. Please let me know if it looks good, or if you have some sample text that I can follow (or copy/paste) :D

File: lib/ui/window.dart [hunk 2]
```diff
@@ -515,8 +515,15 @@ class Locale {
...
 ///
+/// It exposes the size of the display, the core scheduler API, the input event
+/// callback, the graphics drawing API, and other such core services.
+///
 /// There is a single Window instance in the system, which you can
-/// obtain from the [window] property.
+/// obtain from `WidgetsBinding.instance.window`.
+///
+/// There is also a [window] singleton object in `dart:ui` if `WidgetsBinding`
+/// is unavailable. But we strongly advice to avoid statically referencing it.
```
Comment (matthew-carroll) [inline lib/ui/window.dart]: I think this update looks better. Thanks. One last detail, "advice" -> "advise".

Comment (liyuqian) [reply lib/ui/window.dart]: Thank you! Done.

**Pull request merged**